### ISSUE TITLE
Cyclic import in xmipp3 plugin

### ISSUE THREAD

**Issue by delarosatrevin** [scipion-em/scipion-em-xmipp3#5]
@pconesa , @dmaluenda We have a problem here with a cyclic import. 
It is OK from "viewers" to import "protocols" because they are suppose to visualize them or their products. But it is not the other way around...from protocols code (that should be computing code), the viewers functionality should not be imported. 

In this particular case, "nma" submodule is contained in xmipp/protocols and have code for viewers and protocols that is all imported. I suggest that we comment out nma for continue work and its structure should be carefully revised. It can be under xmipp3 and has its own protocols/viewers/etc. 

This issue is quite critical, because now XmippViewer is in xmipp3 and used from many packages...for example the error below:

('Registering plugin: ', 'xmipp3')
   >>> viewers:  error -> Traceback (most recent call last):
  File "scripts/inspect-plugins.py", line 57, in getSubmodule
    m = importlib.import_module('%s.%s' % (name, subname))
  File "/home/josem/work/development/scipion-devel/software/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-eman2/eman2/viewers.py", line 35, in <module>
    from xmipp3.viewers import XmippViewer
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/viewers/__init__.py", line 27, in <module>
    from .viewer import XmippViewer
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/viewers/viewer.py", line 42, in <module>
    from ..protocols.protocol_cl2d_align import XmippProtCL2DAlign
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/__init__.py", line 29, in <module>
    from .nma import *
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/nma/__init__.py", line 33, in <module>
    from .protocol_structure_mapping import XmippProtStructureMapping
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/nma/protocol_structure_mapping.py", line 45, in <module>
    from ..pdb.protocol_pseudoatoms_base import XmippProtConvertToPseudoAtomsBase
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/pdb/__init__.py", line 31, in <module>
    from .viewer_pseudoatoms import XmippPseudoAtomsViewer
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/pdb/viewer_pseudoatoms.py", line 32, in <module>
    from ...viewers import XmippViewer
ImportError: cannot import name XmippViewer

**Comment by dmaluenda** [comment]
Yes. I realized on that and I updated the imports in the view avoiding to load it from the protocols/__init__.py but from its own file.

Did it here: #6

**Comment by delarosatrevin** [comment]
This is still not fixed. Please run the same command at the top and check that the import works.

 josem  ~  work  development  scipion-devel  1  ./scipion python scripts/inspect-plugins.py eman2

Scipion  (2018-08-02)  (jmrt-pluginization c8f5e19)

>>>>> python  "scripts/inspect-plugins.py" "eman2"
Plugin: eman2
   >>> constants:  loaded
   >>> convert:  loaded
   >>> protocols:  loaded
   >>> wizards:  loaded
   >>> viewers:  error -> Traceback (most recent call last):
  File "scripts/inspect-plugins.py", line 57, in getSubmodule
    m = importlib.import_module('%s.%s' % (name, subname))
  File "/home/josem/work/development/scipion-devel/software/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-eman2/eman2/viewers.py", line 35, in <module>
    from xmipp3.viewers import XmippViewer
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/viewers/__init__.py", line 27, in <module>
    from .viewer import XmippViewer
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/viewers/viewer.py", line 42, in <module>
    from xmipp3.protocols.protocol_cl2d_align import XmippProtCL2DAlign
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/__init__.py", line 29, in <module>
    from .nma import *
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/nma/__init__.py", line 33, in <module>
    from .protocol_structure_mapping import XmippProtStructureMapping
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/nma/protocol_structure_mapping.py", line 45, in <module>
    from ..pdb.protocol_pseudoatoms_base import XmippProtConvertToPseudoAtomsBase
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/pdb/__init__.py", line 31, in <module>
    from .viewer_pseudoatoms import XmippPseudoAtomsViewer
  File "/home/josem/work/development/scipion-em-plugins/scipion-em-xmipp3/xmipp3/protocols/pdb/viewer_pseudoatoms.py", line 32, in <module>
    from xmipp3.viewers import XmippViewer
ImportError: cannot import name XmippViewer

**Comment by dmaluenda** [comment]
Yes yes. I first thought that it was fixed and I commented, but then I saw that it's not the same and I deleted my comment, however the reference issue->PR cannot be deleted...
